.github/workflows/publish_images.yaml

```diff
@@ -7,25 +7,43 @@ on:
   schedule:
     # * is a special character in YAML so you have to quote this string
     - cron: '0 1 * * *'
+  workflow_dispatch:
+    inputs:
+      pulpcore:
+        description: 'Pulpcore Version (e.g. 3.7)'
+        required: true
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Build the images
+      - name: Download s6-overlay
         run: |
           wget https://github.com/just-containers/s6-overlay/releases/download/v1.22.1.0/s6-overlay-amd64.tar.gz
-          docker build --file pulp_ci/Containerfile --tag pulp/pulp-ci:latest .
-          docker build --file pulp_fedora31/Containerfile --tag pulp/pulp-fedora31:latest .
-          docker build --file pulp_galaxy_ng/Containerfile --tag pulp/pulp-galaxy-ng:latest .
+      - name: Build the images
+        run: |
+          version="${{ github.event.inputs.pulpcore }}"
+          if [ -z "$version" ]; then
+            version="latest"
+            docker build --file pulp_ci/Containerfile --tag pulp/pulp-ci:latest .
+            docker build --file pulp_fedora31/Containerfile --tag pulp/pulp-fedora31:latest .
+            docker build --file pulp_galaxy_ng/Containerfile --tag pulp/pulp-galaxy-ng:latest .
+          else
+            docker build --file pulp_fedora31/Containerfile --tag pulp/pulp-fedora31:$version --build-arg PULPCORE_VERSION="==$version" .
+            docker build --file pulp_galaxy_ng/Containerfile --tag pulp/pulp-galaxy-ng:$version --build-arg PULPCORE_VERSION="==$version" .
```

Comment:
Can we make this in a way that pulp-fedora31:3.7 will install pulpcore 3.7.3?
Does pip understand `pulpcore~=3.7`?

Reply:
It does although we'd want `pulpcore~=3.7.0` for pulpcore-fedora1:3.7. I updated the workflow.